script/.build-rpm/criu.spec

```diff
@@ -0,0 +1,219 @@
+Name: criu
+Version: %{_version}
+Release: 1%{?dist}
+Provides: crtools = %{version}-%{release}
+Obsoletes: crtools <= 1.0-2
+Summary: Tool for Checkpoint/Restore in User-space
+Group: System Environment/Base
+License: GPLv2
+URL: http://criu.org/
+Source0: http://download.openvz.org/criu/criu-%{version}.tar.bz2
+
+BuildRequires: protobuf-devel protobuf-c-devel python2-devel
+%if 0%{?fedora}
+BuildRequires: asciidoc xmlto
+%endif
+
+# user-space and kernel changes are only available for x86_64 and ARM
+# code is very architecture specific
+# once imported in RCS it needs a bug openend explaining the ExclusiveArch
+# https://bugzilla.redhat.com/show_bug.cgi?id=902875
+ExclusiveArch: x86_64 %{arm}
+
+%description
+criu is the user-space part of Checkpoint/Restore in User-space
+(CRIU), a project to implement checkpoint/restore functionality for
+Linux in user-space.
+
+%package devel
+Summary: Header files and libraries for %{name}
+Group: Development/Libraries
+Requires: %{name} = %{version}-%{release}
+
+%description devel
+This package contains header files and libraries for %{name}.
+
+%package -n python-%{name}
+Summary: Python bindings for %{name}
+Group: Development/Languages
+Requires: %{name} = %{version}-%{release} python-ipaddr protobuf-python
+
+%description -n python-%{name}
+python-%{name} contains Python bindings for %{name}.
+
+%package -n crit
+Summary: CRIU image tool
+Requires: python-%{name} = %{version}-%{release}
+
+%description -n crit
+crit is a tool designed to decode CRIU binary dump files and show
+their content in human-readable form.
+
+
+%prep
+%setup -q -n criu-%{version}
+
+%build
+# %{?_smp_mflags} does not work
+# -fstack-protector breaks build
+CFLAGS+=`echo %{optflags} | sed -e 's,-fstack-protector\S*,,g'` make V=1 WERROR=0 PREFIX=%{_prefix}
+%if 0%{?fedora}
+make docs V=1
+%endif
+
+
+%install
+mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/logrotate.d
+%if 0%{?rhel}
+# disable documentation as it requires asciidoc (which is not available on RHEL7)
+sed -i -e "s,$(CRIU-LIB) install-man,$(CRIU-LIB),g" Makefile
+%endif
+make install DESTDIR=$RPM_BUILD_ROOT PREFIX=%{_prefix} LIBDIR=%{_libdir} LOGROTATEDIR=%{_sysconfdir}/logrotate.d
+
+%if 0%{?fedora}
+# upstream renamed to binary to criu
+ln -s %{_sbindir}/criu $RPM_BUILD_ROOT%{_sbindir}/crtools
+%endif
+
+%post -p /sbin/ldconfig
+%postun -p /sbin/ldconfig
+
+%files
+%{_sbindir}/%{name}
+%if 0%{?fedora}
+%{_sbindir}/crtools
+%doc %{_mandir}/man8/criu.8*
+%endif
+%{_unitdir}/criu.service
+%{_unitdir}/criu.socket
+%{_libdir}/*.so.*
+%{_sysconfdir}/logrotate.d/%{name}-service
+%doc README.md COPYING
+
+%files devel
+%{_includedir}/criu
+%{_libdir}/*.so
+%{_libdir}/pkgconfig/*.pc
+
+%files -n python-%{name}
+%{python2_sitelib}/pycriu/*
+%{python2_sitelib}/*egg-info
+
+%files -n crit
+%{_bindir}/crit
+
+
+%changelog
```

Comment:
no ya im totally for another route :) and i was hoping you knew of
something

On Thu, Sep 3, 2015 at 5:46 PM, Tianon Gravi <notifications@github.com>
wrote:

> In script/.build-rpm/criu.spec
> <https://github.com/opencontainers/runc/pull/245#discussion_r38712791>:
>
> > +%doc README.md COPYING
> > +
> > +%files devel
> > +%{_includedir}/criu
> > +%{_libdir}/*.so
> > +%{_libdir}/pkgconfig/*.pc
> > +
> > +%files -n python-%{name}
> > +%{python2_sitelib}/pycriu/*
> > +%{python2_sitelib}/*egg-info
> > +
> > +%files -n crit
> > +%{_bindir}/crit
> > +
> > +
> > +%changelog
>
> I mention this because updating our criu packaging here is going to be
> IMO unnecessary churn in the runc repo that's pretty unrelated to runc
> development or even deployment. [image: :confused:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/runc/pull/245/files#r38712791>.
>